test/run/test_exports.py

```diff
@@ -0,0 +1,86 @@
+import contextlib
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+
+import pytest
+
+from .. import test
+
+
+@pytest.fixture(name="jsondata", scope="module")
+def jsondata_fixture():
+    return json.dumps({
+        "version": "2",
+        "pipelines": [
+            {
+                "name": "image",
+                "stages": [
+                    {
+                        "type": "org.osbuild.truncate",
+                        "options": {
+                            "filename": "foo.img",
+                            "size": "10",
+                        },
+                    },
+                    {
+                        # cannot use org.osbuild.chown as it needs the chown
+                        # binary in the stage
+                        "type": "org.osbuild.testing.injectpy",
+                        "options": {
+                            "code": [
+                                'import os',
+                                'os.chown(f"{tree}/foo.img", 1000, 1000)',
+                            ],
+                        },
+                    },
+                ]
+            }
+        ]
+    })
+
+
+@pytest.fixture(name="osb", scope="module")
+def osbuild_fixture():
+    with test.OSBuild() as osb:
+        yield osb
+
+
+@pytest.fixture(name="injectpy_libdir", scope="module")
+def injectpy_libdir_fixture(tmpdir_factory):
+    tests_path = pathlib.Path(__file__).parent.parent
+    inject_py_path = tests_path / "stages/org.osbuild.testing.injectpy"
+    fake_libdir_path = tmpdir_factory.mktemp("fake-libdir")
+    # construct minimal viable libdir from current checkout
+    for p in tests_path.parent.glob("*"):
```

Comment:
Probably, unfortunately we really need to copy more then stages, it needs also runners,  assemblers, schemas. And then it needs an empty `osbuild` dir (which I can probably fix by making buildroot.py:248 smarter). I left it to copy everything in stages/runners/assemblers/schemas but I can change it if you prefer that. Technically we also don't need the assemblers right now but given that the fixture has a relatively generic name now it would be strange to not include it.

Reply:
Right.  Didn't think this through entirely.  That's a lot of stuff.  I feel bad for not just going with the first way now.

Reply:
No worries, I think the end result is nicer so it was worth it!